File: src/main/java/me/ksviety/teleporter/teleporters/EntityTeleporter.java
package me.ksviety.teleporter.teleporters;

import me.ksviety.teleporter.PositionProvider;
import net.minecraft.entity.Entity;
import net.minecraft.server.MinecraftServer;
import net.minecraft.util.math.Vec3i;
import net.minecraftforge.fml.common.FMLCommonHandler;

public class EntityTeleporter implements EntityTeleporter {
    private final PositionProvider positionProvider;

    public EntityTeleporter(PositionProvider positionProvider) {
        this.positionProvider = positionProvider;
    }

    @Override
    public void teleport(Entity entity) {
        Vec3i position = positionProvider.provide();

        final MinecraftServer server = FMLCommonHandler.instance().getMinecraftServerInstance();

        server.commandManager.executeCommand(
                server,
                String.format("/tp %s %d %d %d", entity.getName(), position.getX(), position.getY(), position.getZ())
        );
    }
}

File: src/main/kotlin/me/ksviety/teleporter/teleporters/EntityTeleporter.kt
package me.ksviety.teleporter.teleporters

import me.ksviety.teleporter.Position
import me.ksviety.teleporter.Teleporter
import net.minecraft.entity.Entity

class EntityTeleporter(private val position: Position) : Teleporter<Entity> {

    override fun teleport(obj: Entity) {
        val position = position.getValue()

        val x = position.x.toDouble()
        val y = position.y.toDouble()
        val z = position.z.toDouble()

        obj.setPositionAndUpdate(x, y, z)
    }
}
